src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+  const modifiedObject = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(modifiedObject, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach(key => {
+          if (modifiedObject.hasOwnProperty(key)) {
+            delete modifiedObject[key];
+          }
```

Comment:
You dont need 'if ' statement, just leave  'delete modifiedObject[key]' and thats gona work